File: library/src/main/java/me/shkschneider/skeleton/java/EnumHelper.java
package me.shkschneider.skeleton.java;

import java.security.SecureRandom;

public class EnumHelper {

    // <http://stackoverflow.com/a/14257525>
    public static <T extends Enum<?>> T random(final Class<T> cls) {
        return cls.getEnumConstants()[new SecureRandom().nextInt(cls.getEnumConstants().length)];
    }

}

File: library/src/main/java/me/shkschneider/skeleton/java/EnumHelper.kt
package me.shkschneider.skeleton.java

import java.security.SecureRandom

object EnumHelper {

    // <http://stackoverflow.com/a/14257525>
    fun <T : Enum<Any>> random(cls: Class<T>): T {
        return cls.enumConstants[SecureRandom().nextInt(cls.enumConstants.size)]
    }

}
